### PR TITLE
Pull request after book

### PR DESCRIPTION
**Pull request by Tomaszuniek**

### CODE REVIEW

File: demo/src/main/java/Functions/GenerateCSVFunctions.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.List;
+import static com.example.demo.randomvaluesfunctions.RandomValuesFunctions.generateLocation;
+
+public class CSVGenerator {
+    public static List<Location> generateJSON(List<Location> locations, int size){
+        locations.clear();
+        for(int i=0; i<size; i++) {
+            locations.add(generateLocation());
+        };
+        return (locations);
```
Comment (kamrar) [inline demo/src/main/java/Functions/GenerateCSVFunctions.java]: po co te nawiasy ?

File: demo/src/main/java/Functions/GenerateCSVFunctions.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public static String[] generateSeparatedStrings(String unseparatedFields){
+        unseparatedFields = unseparatedFields.replaceAll("\\s","");
+        String[] separatedRequestedFields = unseparatedFields.split(",");
+        return separatedRequestedFields;
```
Comment (kamrar) [inline demo/src/main/java/Functions/GenerateCSVFunctions.java]: zastanow sie nad cialem tej metody i powiedz mi czy mozna to uproscic

File: demo/src/main/java/Functions/GenerateCSVFunctions.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            for(String requestedField : separatedRequestedFields) {
+                csv += returnFieldValue(location, requestedField);
+            }
+            csv += "\n";
+        }
+        return csv;
+    }
+
+    public static String returnFieldValue(Location location, String requestedField) {
+        switch (requestedField) {
```
Comment (kamrar) [inline demo/src/main/java/Functions/GenerateCSVFunctions.java]: czy nie byloby prosciej return location.get(requestedField) + "," ?

File: demo/src/main/java/Functions/GenerateCSVFunctions.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            case "longitude":
+                return location.getLongitude() + ",";
+            default:
+                throw new IllegalArgumentException("field does not exist");
+        }
+    }
+
+    public static String generateCsvFixedFields(List<Location> locations){
+        String csv = "type" + "," + "_id" + "," + "name" + "," + "latitude" + "," + "longitude" + "," + "\n";
+        for (Location lokacja : locations) {
```
Comment (kamrar) [inline demo/src/main/java/Functions/GenerateCSVFunctions.java]: lokacja to po polsku :P

File: demo/src/main/java/Functions/RandomValuesFunctions.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+public class RandomValuesFunctions {
+	public static Location generateLocation() {
+		String country = generateString();
+		String name = generateString();
+		return new Location(generateString(), generateInt(1000000), generateString(), name, name + ", " + country, generateString(), generateString(), country, generateMap(), generateInt(100000), generateBool(), generateString(), generateBool(), generateInt(1000000));
+	}
+
+	public static String generateString() {
+		int leftLimit = 97;
```
Comment (kamrar) [inline demo/src/main/java/Functions/RandomValuesFunctions.java]: a tu co sie stanelo ?

File: demo/src/main/java/Functions/RandomValuesFunctions.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	    int rightLimit = 122; 
+	    Random random = new Random();
+	    int targetStringLength = random.nextInt(10)+1;
+	    StringBuilder buffer = new StringBuilder(targetStringLength);
```
Comment (kamrar) [inline demo/src/main/java/Functions/RandomValuesFunctions.java]: stringbuilder to nie buffer

File: demo/src/main/java/Functions/RandomValuesFunctions.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	    for (int i = 0; i < targetStringLength; i++) {
+	        int randomLimitedInt = leftLimit + (int)
```
Comment (kamrar) [inline demo/src/main/java/Functions/RandomValuesFunctions.java]: bardzo nieczytelne, juz lepiej w jednej linii

File: demo/src/main/java/Functions/RandomValuesFunctions.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	    Random random = new Random();
+	    int targetStringLength = random.nextInt(10)+1;
+	    StringBuilder buffer = new StringBuilder(targetStringLength);
+	    for (int i = 0; i < targetStringLength; i++) {
+	        int randomLimitedInt = leftLimit + (int) 
+	          (random.nextFloat() * (rightLimit - leftLimit + 1));
+	        buffer.append((char) randomLimitedInt);
+	    }
+	    String generatedString = buffer.toString();
+	    return(generatedString);
```
Comment (kamrar) [inline demo/src/main/java/Functions/RandomValuesFunctions.java]: a tu sie wydazylo

**Comment by Tomaszuniek** [comment]
Hej Kamil,dzieki za uwagi, zapamietam i zaraz poprawie.Probowalem zbuildowac tego travisa ale cały czas mam błędy, zrobiłem forka Twojego GJMH i też mi się nie builduje, Tobie działa?Poprawiam teraz z Twoimi uwagami i czytam dalej książkę (klasy).Do jutra,TomekOd: "Kamil Raróg" &lt;notifications@github.com&gt;Do: "Tomaszuniek/zadankodlakamila" &lt;zadankodlakamila@noreply.github.com&gt;; Wysłane: 21:15 Piątek 2020-02-28Temat: Re: [Tomaszuniek/zadankodlakamila] Pull request after book (#4)@kamrar requested changes on this pull request.

In demo/src/main/java/Functions/GenerateCSVFunctions.java:
&gt; @@ -0,0 +1,82 @@
+package ourFunctions;
+
+import com.example.demo.location.Location;
+import java.util.List;
+import static com.example.demo.randomvaluesfunctions.RandomValuesFunctions.generateLocation;
+
+public class CSVGenerator {
+    public static List&lt;Location&gt; generateJSON(List&lt;Location&gt; locations, int size){
+        locations.clear();
+        for(int i=0; i&lt;size; i++) {
+            locations.add(generateLocation());
+        };
+        return (locations);

po co te nawiasy ?

In demo/src/main/java/Functions/GenerateCSVFunctions.java:
&gt; +import java.util.List;
+import static com.example.demo.randomvaluesfunctions.RandomValuesFunctions.generateLocation;
+
+public class CSVGenerator {
+    public static List&lt;Location&gt; generateJSON(List&lt;Location&gt; locations, int size){
+        locations.clear();
+        for(int i=0; i&lt;size; i++) {
+            locations.add(generateLocation());
+        };
+        return (locations);
+    }
+
+    public static String[] generateSeparatedStrings(String unseparatedFields){
+        unseparatedFields = unseparatedFields.replaceAll("\\s","");
+        String[] separatedRequestedFields = unseparatedFields.split(",");
+        return separatedRequestedFields;

zastanow sie nad cialem tej metody i powiedz mi czy mozna to uproscic

In demo/src/main/java/Functions/GenerateCSVFunctions.java:
&gt; +    public static String returnRequestedFieldValues(List&lt;Location&gt; locations, String[] separatedRequestedFields) {
+        String csv = "";
+        for (String requestedField : separatedRequestedFields) {
+            csv += requestedField + ",";
+        }
+        for (Location location : locations) {
+            for(String requestedField : separatedRequestedFields) {
+                csv += returnFieldValue(location, requestedField);
+            }
+            csv += "\n";
+        }
+        return csv;
+    }
+
+    public static String returnFieldValue(Location location, String requestedField) {
+        switch (requestedField) {

czy nie byloby prosciej return location.get(requestedField) + "," ?

In demo/src/main/java/Functions/GenerateCSVFunctions.java:
&gt; +            case "coreCountry":
+                return location.isCoreCountry() + ",";
+            case "distance":
+                return location.getDistance() + ",";
+            case "latitude":
+                return location.getLatitude() + ",";
+            case "longitude":
+                return location.getLongitude() + ",";
+            default:
+                throw new IllegalArgumentException("field does not exist");
+        }
+    }
+
+    public static String generateCsvFixedFields(List&lt;Location&gt; locations){
+        String csv = "type" + "," + "_id" + "," + "name" + "," + "latitude" + "," + "longitude" + "," + "\n";
+        for (Location lokacja : locations) {

lokacja to po polsku :P

In demo/src/main/java/Functions/RandomValuesFunctions.java:
&gt; +
+import com.example.demo.location.Location;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.TreeMap;
+
+public class RandomValuesFunctions {
+	public static Location generateLocation() {
+		String country = generateString();
+		String name = generateString();
+		return new Location(generateString(), generateInt(1000000), generateString(), name, name + ", " + country, generateString(), generateString(), country, generateMap(), generateInt(100000), generateBool(), generateString(), generateBool(), generateInt(1000000));
+	}
+
+	public static String generateString() {
+		int leftLimit = 97;

a tu co sie stanelo ?

In demo/src/main/java/Functions/RandomValuesFunctions.java:
&gt; +import java.util.Random;
+import java.util.TreeMap;
+
+public class RandomValuesFunctions {
+	public static Location generateLocation() {
+		String country = generateString();
+		String name = generateString();
+		return new Location(generateString(), generateInt(1000000), generateString(), name, name + ", " + country, generateString(), generateString(), country, generateMap(), generateInt(100000), generateBool(), generateString(), generateBool(), generateInt(1000000));
+	}
+
+	public static String generateString() {
+		int leftLimit = 97;
+	    int rightLimit = 122; 
+	    Random random = new Random();
+	    int targetStringLength = random.nextInt(10)+1;
+	    StringBuilder buffer = new StringBuilder(targetStringLength);

stringbuilder to nie buffer

In demo/src/main/java/Functions/RandomValuesFunctions.java:
&gt; +
+public class RandomValuesFunctions {
+	public static Location generateLocation() {
+		String country = generateString();
+		String name = generateString();
+		return new Location(generateString(), generateInt(1000000), generateString(), name, name + ", " + country, generateString(), generateString(), country, generateMap(), generateInt(100000), generateBool(), generateString(), generateBool(), generateInt(1000000));
+	}
+
+	public static String generateString() {
+		int leftLimit = 97;
+	    int rightLimit = 122; 
+	    Random random = new Random();
+	    int targetStringLength = random.nextInt(10)+1;
+	    StringBuilder buffer = new StringBuilder(targetStringLength);
+	    for (int i = 0; i &lt; targetStringLength; i++) {
+	        int randomLimitedInt = leftLimit + (int) 

bardzo nieczytelne, juz lepiej w jednej linii

In demo/src/main/java/Functions/RandomValuesFunctions.java:
&gt; +		return new Location(generateString(), generateInt(1000000), generateString(), name, name + ", " + country, generateString(), generateString(), country, generateMap(), generateInt(100000), generateBool(), generateString(), generateBool(), generateInt(1000000));
+	}
+
+	public static String generateString() {
+		int leftLimit = 97;
+	    int rightLimit = 122; 
+	    Random random = new Random();
+	    int targetStringLength = random.nextInt(10)+1;
+	    StringBuilder buffer = new StringBuilder(targetStringLength);
+	    for (int i = 0; i &lt; targetStringLength; i++) {
+	        int randomLimitedInt = leftLimit + (int) 
+	          (random.nextFloat() * (rightLimit - leftLimit + 1));
+	        buffer.append((char) randomLimitedInt);
+	    }
+	    String generatedString = buffer.toString();
+	    return(generatedString);

a tu sie wydazylo

**Pull request merged**